### PR TITLE
add loss module + test loss

### PR DESCRIPTION
**Pull request by QuentinGABOT**

### CODE REVIEW

File: src/torchcvnn/nn/modules/loss.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import torch
+import torch.nn as nn
+
+
+class ComplexMeanSquareError(nn.Module):
```
Comment (jeremyfix) [inline src/torchcvnn/nn/modules/loss.py:30]: I suggest we keep it named as in upstream torch , as   ComplexMSELoss

File: src/torchcvnn/nn/modules/loss.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    def __init__(self):
+        super(ComplexMeanSquareError, self).__init__()
```
Comment (jeremyfix) [inline src/torchcvnn/nn/modules/loss.py:32]: There is no need to define the constructor if it is just the default constructor

However, I would suggest to keep the same arguments than pytorch MSELoss:

```
    def __init__(self, size_average=None, reduce=None, reduction: str = 'mean') -> None:
        super().__init__(size_average, reduce, reduction)
```

File: src/torchcvnn/nn/modules/loss.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    def forward(self, y_true, y_pred):
```
Comment (jeremyfix) [inline src/torchcvnn/nn/modules/loss.py:34]: I would keep the same ordering than pytorch : y_pred , y_true     rather than   y_true, y_pred    even if the computation is the same.

File: src/torchcvnn/nn/modules/loss.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        # Ensure y_pred and y_true are torch tensors
+        y_pred = torch.as_tensor(y_pred)
+        y_true = torch.as_tensor(y_true)
```
Comment (jeremyfix) [inline src/torchcvnn/nn/modules/loss.py:37]: I would probably request the input to be tensors rather than converting them ;

File: src/torchcvnn/nn/modules/loss.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+        # Handle complex pred but real true
+        if torch.is_complex(y_pred) and not torch.is_complex(y_true):
+            y_true = torch.complex(y_true, torch.zeros_like(y_true))
```
Comment (jeremyfix) [inline src/torchcvnn/nn/modules/loss.py:41]: I do not think you need this explicit conversion ; 

Substracting a complex valued tensor from a real valued tensor seems to be correctly a complex valued tensor

```
>>> z1 = torch.tensor([1+1j, 0])
>>> z2 = torch.tensor([1, 0])
>>> (z1 - z2).dtype
torch.complex64
>>> (z2 - z1).dtype
torch.complex64
```

File: src/torchcvnn/nn/modules/loss.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+        # Ensure y_true is the same dtype as y_pred
+        y_true = y_true.type_as(y_pred)
```
Comment (jeremyfix) [inline src/torchcvnn/nn/modules/loss.py:44]: This one seems also unnecessary to me

**Review by jeremyfix — CHANGES_REQUESTED** [review]
